libphx/src/shader.rs

```diff
@@ -68,13 +70,19 @@ unsafe fn CreateGLShader(src: *const libc::c_char, type_0: gl::types::GLenum) ->
 
     /* Check for compile errors. */
     let mut status: i32 = 0;
+
     gl::GetShaderiv(this, gl::COMPILE_STATUS, &mut status);
+
     if status == 0 {
         let mut length: i32 = 0;
+
         gl::GetShaderiv(this, gl::INFO_LOG_LENGTH, &mut length);
-        let infoLog: *mut libc::c_char = MemAllocZero((length + 1) as usize) as *mut libc::c_char;
+
+        let infoLog = MemAllocZero((length + 1) as usize) as *mut libc::c_char;
+
```

Comment:
Could we create a CString and resize it to `length+1` elements or something, to avoid the memalloc? Also, this will leak on line 86.

Reply:
It won't reach line 86 because CFatal never returns.

Using CString I'd postpone to another PR

Reply:
Oh thats a very good point, fair enough!